tests/foreman/cli/test_usergroup.py

```diff
@@ -0,0 +1,510 @@
+    def test_positive_create_with_usergroup_name(self):
+        """Create new user group using another user group attached to the
+        initial group. Use user group name as a parameter
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        for name in valid_data_list():
+            with self.subTest(name):
+                sub_user_group = make_usergroup({'name': name})
+                user_group = make_usergroup({
+                    'user-groups': sub_user_group['name']})
+                self.assertEqual(user_group['user-groups'][0], name)
+
+    @tier1
+    def test_positive_create_with_usergroup_id(self):
+        """Create new user group using another user group attached to the
+        initial group. Use user group id as a parameter
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully.
+        """
+        sub_user_group = make_usergroup()
+        user_group = make_usergroup({
+            'user-group-ids': sub_user_group['id']})
+        self.assertEqual(user_group['user-groups'][0], sub_user_group['name'])
+
+    @tier1
+    def test_positive_create_with_usergroups(self):
+        """Create new user group using multiple user groups attached to that
+        initial group. Use user groups name as a parameter
+
+        @Feature: Usergroup
+
+        @Assert: User group is created successfully and contains all expected
+        user groups
+        """
+        sub_user_groups = [make_usergroup()['name'] for _ in range(3)]
+        user_group = make_usergroup({'user-groups': sub_user_groups})
+        for sub_user_group in user_group['user-groups']:
+            self.assertIn(sub_user_group, sub_user_groups)
+
+    @tier1
+    def test_negative_create_with_name(self):
+        """Attempt to create user group with invalid name.
+
+        @Feature: Usergroup
+
+        @Assert: User group is not created.
+        """
+        for name in invalid_values_list():
+            with self.subTest(name):
+                with self.assertRaises(CLIFactoryError):
+                    make_usergroup({'name': name})
+                with self.assertRaises(CLIReturnCodeError):
+                    UserGroup.info({'name': name})
+
+    @tier1
+    def test_negative_create_with_same_name(self):
+        """Attempt to create user group with a name of already existent entity.
+
+        @Feature: Usergroup
+
+        @Assert: User group is not created.
+        """
+        user_group = make_usergroup()
+        with self.assertRaises(CLIFactoryError):
+            make_usergroup({'name': user_group['name']})
+
+    @tier1
+    def test_positive_list(self):
+        """Test user group list command
+
+        @Feature: Usergroup
+
+        @Assert: User group list command returns valid and expected data
+
+        """
+        user_group = make_usergroup()
+        result_list = UserGroup.list({
+            'search': 'name=%s' % user_group['name']})
```

Comment:
I see format being used more than string interpolation, feel free to follow that or leave this way.